### PR TITLE
Issue/7166 update order list automatically

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #7166

**We can merge this PR after review, but we need to make sure to merge https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2959 only when the M2 of Tablet support is ready to be merged into trunk to ensure FluxC's trunk and WooAndroid's trunk remain compatible.**

### Description
This PR removes manual refreshes/reloads of the OrderList list and relies on the ListStore component to update the data source automatically.

It also updates FluxC version which contains significant changes - the ListStore component is now informed about any changes to the OrderModel in the database so it can update the data source accordingly. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2959

This PR contains one regression - not sure if it's a regression to be honest, however, when filters are changed, the order detail in two pane layout isn't automatically updated -> the user needs to select the item they want to open. Considering opening an order detail sends several remote requests I'm not sure this is a true regression - felt quite natural to me.

### Testing instructions

Features to test
- Create order and ensure the new item is available in the list and auto-selected
- Update order status or total price and ensure the list item is automatically updated
- Change filters and ensure they work as expected
- Use search and ensure it works as expcted

### Images/gif

- The only UI facing change is that the list updates are now smoother and faster.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt [hunk 1]
```diff
@@ -462,8 +458,6 @@ class OrderListFragment :
                             openFirstOrder()
                         }, HANDLER_DELAY)
                     }
-
-                    viewModel.viewState.filterCount > 0 -> openFirstOrder()
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt:466]: This is the line which can be considered as regression - we don't open the first item after a filter is changed.

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt:466]: We could potentially check if the selected item is available in the list and invoke `openFirstOrder` only when it isn't. Something like 

```
viewModel.viewState.filterCount > 0 &&
                        pagedOrdersList
                            .find { 
                                (it as? OrderListItemUIType.OrderListItemUI)?.orderId == selectedOrder.selectedOrderId.value } != null
```

but this code feels so hard to maintain that I don't believe it's worth it. I think a better approach would be to implement some more robust solution which would take care of syncing the list with teh order detail and would work for both tablets and phones consistently.

**Review by samiuelson — COMMENTED** [review]
Thanks, @malinajirka, the code looks good to me overall 👍🏼 

I confirmed that the order list gets updated both on tablet and phone after adding a new order, updating status and applying filters.

How about issuing a refund, are these PRs expected to implement auto update of the order list in this case as well? It's not working for me:

https://github.com/woocommerce/woocommerce-android/assets/4527432/6eb485b3-26db-42a8-af87-aad89cec0c57

**Comment by malinajirka** [comment]
Thanks for the review @samiuelson !

> How about issuing a refund, are these PRs expected to implement auto update of the order list in this case as well? It's not working for me:

It is supposed to handle all order updates 🤔 . I tested the refund flow quite a bit and it worked every time for me. I just re-tested it and still works. 😕 

[Screen_recording_20240222_134436.webm](https://github.com/woocommerce/woocommerce-android/assets/2261188/4cb6c3c5-fc29-40dd-9bde-b2eb33981265)

Any ideas what might be different in our setup?

**Review by samiuelson — APPROVED** [review]
Refund seems to be working well after updating to the latest FluxC version.

**Pull request merged**